packages/teleport/src/Nodes/AddNode/Iam/Iam.tsx

```diff
@@ -0,0 +1,267 @@
+import React from 'react';
+import styled from 'styled-components';
+import { DialogContent, DialogFooter } from 'design/Dialog';
+import {
+  Text,
+  Box,
+  ButtonPrimary,
+  Link,
+  Alert,
+  ButtonLink,
+  ButtonSecondary,
+} from 'design';
+import TextSelectCopy from 'teleport/components/TextSelectCopy';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+import { Rule } from 'teleport/services/joinToken';
+import DownloadLinks from 'teleport/components/DownloadLinks';
+import { Pencil } from 'design/Icon';
+
+export default function Iam({
+  token,
+  expiry,
+  attempt,
+  onGenerate,
+  isEnterprise,
+  version,
+  onClose,
+}: Props) {
+  const { hostname, port } = window.document.location;
+  const host = `${hostname}:${port || '443'}`;
```

Comment:
Do we want to have this default like this? Or should we instead rely on the protocol?

Reply:
Do you mean to use port 80 in case the protocol is `HTTP`? I'm not sure, can you run Teleport without a cert?